python/ray/llm/_internal/serve/builders/application_builders.py

```diff
@@ -22,15 +26,37 @@ def build_llm_deployment(
     name_prefix: Optional[str] = None,
     deployment_kwargs: Optional[dict] = None,
 ) -> Application:
-    name_prefix = name_prefix or "LLMDeployment"
-    deployment_kwargs = deployment_kwargs or {}
+    dp_size = llm_config.engine_kwargs.get("data_parallel_size", 1)
+    if dp_size == 1:
+        name_prefix = name_prefix or "LLMDeployment:"
+        deployment_kwargs = deployment_kwargs or {}
 
-    deployment_options = llm_config.get_serve_options(
-        name_prefix=name_prefix,
-    )
+        deployment_options = llm_config.get_serve_options(
+            name_prefix=name_prefix,
+        )
+        return LLMDeployment.options(**deployment_options).bind(
+            llm_config=llm_config, **deployment_kwargs
+        )
+
+    # Build data parallel LLM deployment.
+    dp_rank_assigner = DPRankAssigner.bind(dp_size=dp_size)
+    name_prefix = name_prefix or "DPLLMDeployment:"
+    name = name_prefix + llm_config._get_deployment_name()
+    num_replicas = llm_config.deployment_config.get("num_replicas", None)
```

Comment:
checking `num_replicas` is not the most robust way to avoid undefined behavior. To completely avoid undefined behavior let's do this for now: 

Error if `num_replicas` is set. 
Error if `autoscaling_config` is set.